Core/src/IfcGeometryConverter/PlacementConverter.h

```diff
@@ -35,1657 +35,1664 @@
+                            //applyContext( context, context_matrix, length_factor, placement_already_applied );
+                            //object_placement_matrix = context_matrix*object_placement_matrix;
+                        }
+                    }
+
+
+                    /*! \brief Converts \c IfcLocalPlacement to a transformation matrix.
+
+                    \param	alreadyApplied		An array of references to already applied \c IfcObjectPlacement-s.
+                    \param[in]	local_placement		\c IfcLocalPlacement entity to be interpreted.
+                    \param[out] matrix			Calculated transformation matrix.
+
+                    Function computes the local placement and stores it in the matrix.
+                    */
+                    void convertIfcLocalPlacement(std::vector<std::shared_ptr<typename IfcEntityTypesT::IfcObjectPlacement>>& alreadyApplied, carve::math::Matrix& object_placement_matrix, std::shared_ptr<typename IfcEntityTypesT::IfcLocalPlacement> local_placement)
+                    {
+                        // **************************************************************************************************************************
+                        //  https://standards.buildingsmart.org/IFC/RELEASE/IFC4_1/FINAL/HTML/link/ifclocalplacement.htm
+                        // ENTITY IfcLocalPlacement
+                        //	SUBTYPE OF(IfcObjectPlacement);
+                        //		PlacementRelTo: OPTIONAL IfcObjectPlacement; // was promoted to ObjectPlacement from IFC4x2+
+                        //		RelativePlacement: IfcAxis2Placement;
+                        //	WHERE
+                        //		WR21 : IfcCorrectLocalPlacement(RelativePlacement, PlacementRelTo);
+                        // END_ENTITY;
+                        // **************************************************************************************************************************
+
+                        // RelativePlacement				
+                        // TYPE IfcAxis2Placement = SELECT (
+                        //	IfcAxis2Placement2D,
+                        //	IfcAxis2Placement3D);
+                        // END_TYPE;
+                        convertRelativePlacement(object_placement_matrix, local_placement);
+                        carve::math::Matrix relative_placement(carve::math::Matrix::IDENT());
+                        convertIfcLocalPlacementRelationPoint(alreadyApplied, local_placement, relative_placement);
+                        object_placement_matrix = relative_placement * object_placement_matrix;
+
+                    }
+
+                    void convertRelativePlacementOriginInIfcLinearPlacement(std::vector<std::shared_ptr<typename IfcEntityTypesT::IfcObjectPlacement>>& alreadyApplied, carve::math::Matrix& object_placement_matrix, std::shared_ptr<typename IfcEntityTypesT::IfcLinearPlacement> linear_placement);

```

Comment:
Doxycomment missing. There is a problem as well, as a relative linear placement need to be handled totally differently than a relative local placement. Needs further investigation.

Reply:
Done